compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -145,7 +146,11 @@ pub struct Parser<'a> {
 
     /// This allows us to recover when the user forget to add braces around
     /// multiple statements in the closure body.
-    pub last_closure_body: Option<Span /* The closing `|` of the closure declarator. */>,
+    pub last_closure_body: Option<(
```

Comment:
Given that it's no longer just the body, we should rename this to `last_closure` or `current_closure`.